parlai/core/metrics.py

```diff
@@ -251,6 +251,165 @@ def value(self) -> float:
         return self._numer / self._denom
 
 
+class PrecisionMetric(Metric):
```

Comment:
i don't think the confusion matrix should be tracked by `PrecisionMetric`. rather we should have something generic like `ConfusionMatrixMetric` or `ClassificationMetrics` that keep track of the confusion matrix, and then return prec, recall, and f1 when we call `compute_many`.